eli5/formatters/html.py

```diff
@@ -119,12 +120,14 @@ def render_targets_weighted_spans(targets, preserve_density):
 
 def render_weighted_spans(pws):
     # type: (PreparedWeightedSpans) -> str
-    # TODO - can be much smarter, join spans at least
-    # TODO - for longer documents, remove text without active features
+    # TODO - for longer documents, an remove text without active features
     return ''.join(
-        _colorize(token, weight, pws.weight_range)
-        for token, weight in zip(
-            pws.doc_weighted_spans.document, pws.char_weights))
+        _colorize(''.join(t for t, _ in tokens_weights),
+                  weight,
+                  pws.weight_range)
+        for weight, tokens_weights in groupby(
+            zip(pws.doc_weighted_spans.document, pws.char_weights),
```

Comment:
Hm, I think it's a bit trickier than that: while we show weight with just 3 significant figures, it's an "absolute" weight, not relative, and color is chosen relative to weight range. So it can be that both weight and weight range are small. And it's also not quite correct that we always show 3 digits for weights in title.